### DISCUSSION TITLE
[Compilers] `isSystem` flag is passed to the compiler, enables ALL calls to have the `isSystem` simulation?

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [zkSync-Community-Hub/zksync-developers#491] [Compilers] [answered]
### Which compiler are you using?

- [X] zksolc
- [ ] zkvyper

### Which version of Solidity or Vyper?

0.8.24

### zksolc/zkvyper version?

NA

### Explain the issue

```bash
Just to inform everyone, I have updated the use of the `isSystem` flag here. 

https://ethereum.stackexchange.com/questions/162863/how-does-the-issystem-flag-work-for-zksync-projects/163041#163041

Someone (maybe me?) should make a PR to the system.
```

**Comment by bxpana** [comment, marked as answer]
This has been responded to here with the answer:
https://ethereum.stackexchange.com/questions/162863/how-does-the-issystem-flag-work-for-zksync-projects/163041#163041
